example/src/screens/GenericClientRequestScreen.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState } from "react";
+import { Alert, SafeAreaView, ScrollView } from "react-native";
+import { Button, ButtonGroup, Input } from "react-native-elements";
+
+import { Constants } from "@mattermost/react-native-network-client";
+
+import AddHeaders from "../components/AddHeaders";
+import NumericInput from "../components/NumericInput";
+import ResponseOverlay from "../components/ResponseOverlay";
+import RetryPolicyConfiguration from "../components/RetryPolicyConfiguration";
+import { parseHeaders, METHODS } from "../utils";
+
+const GenericClientRequestScreen = ({
+    route,
+}: GenericClientRequestScreenProps) => {
+    const { client } = route.params;
+    const [url, setUrl] = useState("");
+    const [selectedMethodIndex, setSelectedMethodIndex] = useState(0);
+    const [timeoutInterval, setTimeoutInterval] = useState(30);
+    const [body, setBody] = useState('{"login_id":"","password":""}');
+    const [requestHeaders, setRequestHeaders] = useState<Header[]>([]);
+    const [response, setResponse] = useState<ClientResponse>();
+    const [responseVisible, setResponseVisible] = useState(false);
+    const [retryPolicyConfiguration, setRetryPolicyConfiguration] = useState<
+        RetryPolicyConfiguration
+    >({
+        type: undefined,
+        retryLimit: 2,
+        exponentialBackoffBase: 2,
+        exponentialBackoffScale: 0.5,
+    });
+
+    const methods = Object.keys(METHODS);
+
+    const toggleRetryPolicyType = () =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            type: retryPolicyConfiguration.type
+                ? undefined
+                : Constants.EXPONENTIAL_RETRY,
+        });
+    const setRetryLimit = (retryLimit: number) =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            retryLimit,
+        });
+    const setExponentialBackoffBase = (exponentialBackoffBase: number) =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            exponentialBackoffBase,
+        });
+    const setExponentialBackoffScale = (exponentialBackoffScale: number) =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            exponentialBackoffScale,
+        });
```

Comment:
These might be worth extracting out into their own custom hook to reuse across the generic / network screens?